test/extended/olm/olmv1.go

```diff
@@ -417,3 +501,318 @@ func checkFeatureCapability(oc *exutil.CLI) {
+          MAX_LATENCY_THRESHOLD_MS=%d
+          SUCCESS_RATE_THRESHOLD=%.1f  
+          # Install required packages
+          dnf install -y git gcc openssl-devel zlib-devel make
+          # Clone and build wrk2
+          git clone https://github.com/giltene/wrk2.git
+          cd wrk2
+          make
+          # Create a Lua script processing results
+          cat > print_report.lua << 'EOL'
+          
+          function done(summary, latency, requests)
+              -- Print basic stats
+              io.write("\n\n=== LOAD TEST RESULTS ===\n")
+              io.write(string.format("Total requests: %%d\n", summary.requests))
+              io.write(string.format("Socket errors: connect %%d, read %%d, write %%d, timeout %%d\n",
+                  summary.errors.connect, summary.errors.read, summary.errors.write, summary.errors.timeout))
+              io.write(string.format("HTTP errors: %%d\n", summary.errors.status))
+              io.write(string.format("Request rate: %%.2f requests/s\n", summary.requests / summary.duration * 1000000))
+              
+              -- Calculate success rate
+              local total_errors = summary.errors.status + summary.errors.connect + 
+                                  summary.errors.read + summary.errors.write + summary.errors.timeout
+              local success_count = summary.requests - total_errors
+              local success_rate = success_count / summary.requests * 100
+              io.write(string.format("Success rate: %%.2f%%%%\n", success_rate))
+              
+              
+              -- Check if we met our success rate threshold
+              local success_threshold = tonumber(os.getenv("SUCCESS_RATE_THRESHOLD") or 90)
+              io.write("\nThreshold checks:\n")
+              io.write(string.format("  Success rate: %%.2f%%%% (threshold: %%.2f%%%%)\n", 
+                       success_rate, success_threshold))
+              if success_rate < success_threshold then
+                  io.write(string.format("FAILED: Success rate (%%.2f%%%%) below threshold (%%.2f%%%%)\n", 
+                           success_rate, success_threshold))
+                  return
+              end
+              
+              io.write("\nSUCCESS: Success rate criteria met. Check latency in wrk2 output.\n")
+          end
+          EOL
+          # Export environment variables for Lua script
+          export MAX_LATENCY_THRESHOLD_MS=$MAX_LATENCY_THRESHOLD_MS
+          export SUCCESS_RATE_THRESHOLD=$SUCCESS_RATE_THRESHOLD
+          # Run the load test and save output
+          echo "Starting wrk2 load test..."
+          ./wrk -t$CONCURRENT_CONNECTIONS -c$CONCURRENT_CONNECTIONS -d${TEST_DURATION_SECONDS}s -R$REQUEST_RATE -s print_report.lua -L $SERVICE_URL
+          # Extract the 99%% latency value from wrk2 output
+          echo ""
+          echo "Test completed. Checking latency threshold..."
+          LATENCY_MS=$(grep "99.000%%" | awk '{print $2}')
```

Comment:
I'm confused by this line. What input is `grep` operating on here and how?

Reply:
This was a bug because of which the report was printing the wrong p99 latency (and the check was incorrect too). Fixed it, PTAL